packages/config/index.js

```diff
@@ -2,36 +2,77 @@
 
 var path = require('path');
 
-var root = require('pkg-dir').sync(process.cwd());
+var assign = require('deep-assign');
+var root = require('pkg-dir').sync();
 
-var npmConfig = require('./lib/parse-env')('hops');
-if (!Object.keys(npmConfig).length) {
-  try {
-    require.resolve(path.join(root, 'package.json'));
-    npmConfig = require(path.join(root, 'package.json')).config.hops;
-  } catch (_) {}
+var explorer = require('cosmiconfig')('hops', {
+  rcExtensions: true,
+  stopDir: root,
+  sync: true,
+});
+
+function getDefaultConfig() {
+  return {
+    https: false,
+    host: '0.0.0.0',
+    port: 8080,
+    locations: [],
+    basePath: '',
+    assetPath: '',
+    browsers: '> 1%, last 2 versions, Firefox ESR',
+    node: 'current',
+    envVars: { HOPS_MODE: 'dynamic' },
+    moduleDirs: [],
+    appDir: '.',
+    buildDir: 'build',
+    cacheDir: 'node_modules/.cache/hops',
+  };
+}
+
+function applyUserConfig(config) {
+  var result = explorer.load(process.cwd());
```

Comment:
What's the reason for using `process.cwd()` here instead of `root` (which points to the hops application root)?

Reply:
I used it because it most closely resembles the present implementation. I can imagine some (rather weird) monorepo setups in which this might be helpful. I many cases both approaches produce identical values, I guess.